pygmt/params/position.py

```diff
@@ -0,0 +1,209 @@
+"""
+The Position class for positioning GMT embellishments.
+"""
+
+import dataclasses
+from collections.abc import Sequence
+from typing import Literal
+
+from pygmt._typing import AnchorCode
+from pygmt.alias import Alias
+from pygmt.exceptions import GMTValueError
+from pygmt.params.base import BaseParam
+
+
+@dataclasses.dataclass(repr=False)
+class Position(BaseParam):
+    """
+    Class for positioning embellishments on a plot.
+
+    .. figure:: https://docs.generic-mapping-tools.org/dev/_images/GMT_anchor.png
+       :width: 600 px
+       :align: center
+
+       The placement of a GMT embellishment (represented by a green rectangle) in
+       relation to the underlying plot (represented by a bisque rectangle).
+
+    This class provides flexible positioning for GMT embellishments (e.g., logo, scale,
+    rose) by defining a *reference point* on the plot and an *anchor point* on the
+    embellishment. The embellishment is positioned so these two points overlap.
+
+    **Conceptual Model**
+
+    Think of it like dropping an anchor from a boat:
+
+    1. The boat navigates to the *reference point* (a location on the plot)
+    2. The *anchor point* (a specific point on the embellishment) is aligned with the
+       *reference point*
+    3. The embellishment is "dropped" at that position
+
+    **Reference Point**
+
+    The *reference point* can be specified in five different ways using the ``type`` and
+    ``location`` attributes:
+
+    ``type="mapcoords"`` Map Coordinates
+        Use data/geographic coordinates. Specify ``location`` as
+        (*longitude*, *latitude*). Useful when tying the embellishment to a specific
+        geographic location.
+
+        **Example:** ``location=(135, 20), type="mapcoords"``.
+
+    ``type="plotcoords"`` Plot Coordinates
+        Use plot coordinates as distances from the lower-left plot origin. Specify
+        ``location`` as (*x*, *y*) with units (e.g., inches, centimeters, points).
+        Useful for precise layout control.
+
+        **Example:** ``location=("2c", "2.5c"), type="plotcoords"``
+
+    ``type="boxcoords"`` Normalized Coordinates
+        Use normalized coordinates where (0, 0) is the lower-left corner and (1, 1) is
+        the upper-right corner of the bounding box of the current plot. Specify
+        ``location`` as (*nx*, *ny*). Useful for positioning relative to plot dimensions
+        without units.
```

Comment:
It should be noted that, although the GMT documentation says that nx/ny are values in 0-1, but actually other values are also supported. For example:
```bash
gmt begin map
gmt basemap -R0/10/0/5 -JX10c/5c -Baf
gmt logo -Dn-1/-1
gmt end show
```

<img width="2380" height="1174" alt="Image" src="https://github.com/user-attachments/assets/6528974d-f756-429d-9694-2269b8144074" />